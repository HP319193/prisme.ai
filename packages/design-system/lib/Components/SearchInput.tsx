import { Input } from 'antd';
import { SearchOutlined } from '@ant-design/icons';
import { InputHTMLAttributes } from 'react';

export interface SearchInputProps
  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'size'> {}

const SearchInput = ({ ...props }: SearchInputProps) => (
  <Input prefix={<SearchOutlined className="text-gray" />} {...props} />
);

export default SearchInput;
