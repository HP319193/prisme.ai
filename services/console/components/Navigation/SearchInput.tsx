import { CloseCircleOutlined } from '@ant-design/icons';
import { useTranslation } from 'next-i18next';
import { useRef } from 'react';
import MagnifierIcon from '../../icons/magnifier.svgr';

interface SearchInputProps {
  value: string;
  onChange: (v: string) => void;
  onFocus?: () => void;
  placeholder?: string;
}
export const SearchInput = ({
  value,
  onChange,
  onFocus,
  placeholder = 'workspace.search',
}: SearchInputProps) => {
  const { t } = useTranslation('workspaces');
  const ref = useRef<HTMLInputElement>(null);
  return (
    <div className="flex flex-1 relative px-2 border-b-[1px] h-[4rem]">
      <button
        className={`m-4 focus:outline-none ${value ? 'text-accent' : ''}`}
        onClick={() => {
          onFocus && onFocus();
          setTimeout(() => {
            ref?.current?.focus();
          }, 200);
        }}
      >
        <MagnifierIcon
          height="1.2rem"
          width="1.2rem"
          className="text-[1.6rem] mr-2"
        />
      </button>
      <input
        ref={ref}
        className="flex flex-1 outline-none text-accent"
        value={value}
        onChange={({ target: { value } }) => onChange(value)}
        placeholder={t(placeholder)}
      />
      {value && (
        <button
          className="focus:outline-none m-1 hover:text-accent"
          onClick={() => {
            onChange('');
            ref?.current?.focus();
          }}
        >
          <CloseCircleOutlined />
        </button>
      )}
    </div>
  );
};
export default SearchInput;
