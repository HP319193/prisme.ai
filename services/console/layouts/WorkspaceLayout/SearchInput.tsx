import { CloseCircleOutlined, SearchOutlined } from '@ant-design/icons';
import { useTranslation } from 'next-i18next';
import { useRef } from 'react';

interface SearchInputProps {
  value: string;
  onChange: (v: string) => void;
  onFocus?: () => void;
}
export const SearchInput = ({ value, onChange, onFocus }: SearchInputProps) => {
  const { t } = useTranslation('workspaces');
  const ref = useRef<HTMLInputElement>(null);
  return (
    <div className="flex flex-1 relative px-2 border-b-[1px] h-[4rem]">
      <button
        className="m-4 focus:outline-none"
        onClick={() => {
          onFocus && onFocus();
          setTimeout(() => {
            ref?.current?.focus();
          }, 200);
        }}
      >
        <SearchOutlined className="text-[1.6rem]" />
      </button>
      <input
        ref={ref}
        className="flex flex-1 outline-none"
        value={value}
        onChange={({ target: { value } }) => onChange(value)}
        placeholder={t('workspace.search')}
      />
      {value && (
        <button
          className="focus:outline-none m-1"
          onClick={() => {
            onChange('');
            ref?.current?.focus();
          }}
        >
          <CloseCircleOutlined />
        </button>
      )}
    </div>
  );
};
export default SearchInput;
