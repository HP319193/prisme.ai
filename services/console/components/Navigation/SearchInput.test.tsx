import renderer from 'react-test-renderer';
import SearchInput from './SearchInput';

it('should render', () => {
  const root = renderer.create(
    <SearchInput value="foo" onChange={console.log} />
  );
  expect(root.toJSON()).toMatchSnapshot();
});
