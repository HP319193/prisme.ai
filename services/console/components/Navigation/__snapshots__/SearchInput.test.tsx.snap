// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="flex flex-1 relative px-2 border-b-[1px] h-[4rem]"
>
  <button
    className="m-4 focus:outline-none text-accent"
    onClick={[Function]}
  >
    <SvgrURL
      className="text-[1.6rem] mr-2"
      height="1.2rem"
      width="1.2rem"
    />
  </button>
  <input
    className="flex flex-1 outline-none text-accent"
    onChange={[Function]}
    placeholder="workspace.search"
    value="foo"
  />
  <button
    className="focus:outline-none m-1 hover:text-accent"
    onClick={[Function]}
  />
</div>
`;
